src/main/java/edu/ucsb/cs156/happiercows/jobs/UpdateCowHealthJobInd.java

```diff
@@ -0,0 +1,94 @@
+package edu.ucsb.cs156.happiercows.jobs;
+
+import java.util.Optional;
+
+import edu.ucsb.cs156.happiercows.entities.Commons;
+import edu.ucsb.cs156.happiercows.entities.User;
+import edu.ucsb.cs156.happiercows.entities.UserCommons;
+import edu.ucsb.cs156.happiercows.repositories.CommonsRepository;
+import edu.ucsb.cs156.happiercows.repositories.UserCommonsRepository;
+import edu.ucsb.cs156.happiercows.repositories.UserRepository;
+import edu.ucsb.cs156.happiercows.services.jobs.JobContext;
+import edu.ucsb.cs156.happiercows.services.jobs.JobContextConsumer;
+import edu.ucsb.cs156.happiercows.strategies.CowHealthUpdateStrategy;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@AllArgsConstructor
+public class UpdateCowHealthJobInd implements JobContextConsumer {
+
+    @Getter
+    private CommonsRepository commonsRepository;
+    @Getter
+    private UserCommonsRepository userCommonsRepository;
+    @Getter
+    private UserRepository userRepository;
+    @Getter
+    private Long commonsID;
+
+    @Override
+    public void accept(JobContext ctx) throws Exception {
+        ctx.log("Updating cow health...");
+
+       Optional<Commons> commonUpdatedOpt = commonsRepository.findById(commonsID);
+
+        if(commonUpdatedOpt.isPresent()){
+            Commons commons = commonUpdatedOpt.get();
+            ctx.log("Commons " + commons.getName() + ", degradationRate: " + commons.getDegradationRate() + ", carryingCapacity: " + commons.getCarryingCapacity());
+            int numUsers = commonsRepository.getNumUsers(commons.getId()).orElseThrow(() -> new RuntimeException("Error calling getNumUsers(" + commons.getId() + ")"));
+
+            if (numUsers==0) {
+                ctx.log("No users in this commons");
+                ctx.log("Cow health has been updated!");
+                return;
+            }
+
+            int carryingCapacity = commons.getCarryingCapacity();
```

Comment:
This logic was already updated in a previous PR, so this code is already out of date.  This is why it's a bad idea to just copy/paste code; we already have version skew before the code is even merged!   See PR https://github.com/ucsb-cs156-m23/proj-happycows-m23-10am-4/pull/35 for the changes to the corresponding line in `UpdateCowHealthJob`.

At a minimum this file needs to be updated with the parallel changes.  But, probably, we should hold ourselves to the higher standard of factoring out the duplicate code so we don't have to maintain this logic in two places.